chronos/asyncsync.nim

```diff
@@ -71,53 +72,48 @@ proc newAsyncLock*(): AsyncLock =
   # Workaround for callSoon() not worked correctly before
   # getGlobalDispatcher() call.
   discard getGlobalDispatcher()
-  result = new AsyncLock
-  result.waiters = newSeq[Future[void]]()
-  result.locked = false
+  AsyncLock(waiters: newSeq[Future[void]](), locked: false, released: true)
```

Comment:
ideally we'd have a `released` flag that starts with `false`  / the default value for bool - that leads to less bugs generally in nim, would have to change name